tests/platform_tests/cli/test_show_platform_dpu.py

```diff
@@ -0,0 +1,183 @@
+"""
+Tests for the `platform cli ...` commands in DPU 
+"""
+
+import http.client
+import logging
+import pytest
+import time
+import redis
+from tests.common.utilities import wait_until
+from . import util
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.platform.device_utils_dpu import *
+from tests.common.helpers.platform_api import chassis, module
+from tests.platform_tests.api.conftest import platform_api_conn
+from tests.common.devices.sonic import *
+
+pytestmark = [
+    pytest.mark.topology('t1')
+]
+
+def test_midplane_ip(duthosts, enum_rand_one_per_hwsku_hostname, platform_api_conn):
+    """
+    @summary: Verify `Midplane ip address between NPU and DPU`
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    num_modules = int(chassis.get_num_modules(platform_api_conn))
+    ping_count = 0
+    dpu_count = 0
+    ip_address_list = []
+
+    output_dpu_status = duthost.show_and_parse('show chassis module status')
+
+    for index in range(len(output_dpu_status)):
+        parse_output = output_dpu_status[index]
+        if 'DPU' in parse_output['name']:
+            if parse_output['oper-status'] != 'Offline':
+                index = (parse_output['name'])[-1]
+                # Api will always return the ip address as it is statically configured now
+                # Check will be added for ip address once dynamic implementation is in place
+                ip_address_list.append(module.get_midplane_ip(platform_api_conn, index))
+
+    check_dpu_ping_status(duthost, ip_address_list)
```

Comment:
There is no failure check here, if ping fails to the DPU hosts. Also, it is good to check for each DPU and log accordingly to  debug the failure easily.

Reply:
added failure check to it.